docs/api.rst

```diff
@@ -4702,9 +4703,81 @@ Dealing with History Events
+      Host: localhost:5042
+      Content-Type: application/json;charset=UTF-8
+
+      {
+          "entry_type": "history event",
+          "event_identifier": "RE_xxxxxxxxxx",
+          "location": "ethereum",
+          "timestamp": 1569924574,
+          "balance": {"amount": "1.542", "usd_value": "1.675"},
+          "sequence_index": 162,
+          "event_type": "informational",
+          "event_subtype": "approve",
+          "asset": "eip155:1/erc20:0x89d24A6b4CcB1B6fAA2625fE562bDD9a23260359",
+          "location_label": "0x2B888954421b424C5D3D9Ce9bB67c9bD47537d12",
+          "notes": "Approve 1 SAI of 0x2B888954421b424C5D3D9Ce9bB67c9bD47537d12 for spending by 0xdf869FAD6dB91f437B59F1EdEFab319493D4C4cE"
+      }
+
+   .. http:example:: curl wget httpie python-requests
+
+      PUT /api/1/history/events HTTP/1.1
+      Host: localhost:5042
+      Content-Type: application/json;charset=UTF-8
+
+      {
+          "entry_type": "eth block event",
+          "location": "ethereum",
+          "timestamp": 1569924574,
+          "balance": {"amount": "1.542", "usd_value": "1.675"},
+          "block_number": 11,
+          "validator_index": 1,
+          "fee_recipient": "0x2B888954421b424C5D3D9Ce9bB67c9bD47537d12",
+          "is_mev_reward": true
+      }
+
+   .. http:example:: curl wget httpie python-requests
+
+      PUT /api/1/history/events HTTP/1.1
+      Host: localhost:5042
+      Content-Type: application/json;charset=UTF-8
+
+      {
+          "entry_type": "eth deposit event",
+          "timestamp": 1569924574,
+          "balance": {"amount": "1.542", "usd_value": "1.675"},
+          "tx_hash": "0x64f1982504ab714037467fdd45d3ecf5a6356361403fc97dd325101d8c038c4e",
+          "event_identifier": "RE_xxxxxxxxxx",
+          "validator_index": 1,
+          "sequence_index": 162,
+          "depositor": "0x2B888954421b424C5D3D9Ce9bB67c9bD47537d12",
+          "extra_data": {}
+      }
+
+   .. http:example:: curl wget httpie python-requests
+
+      PUT /api/1/history/events HTTP/1.1
+      Host: localhost:5042
+      Content-Type: application/json;charset=UTF-8
+
+      {
+          "entry_type": "eth withdrawal event",
+          "timestamp": 1569924574,
+          "balance": {"amount": "1.542", "usd_value": "1.675"},
+          "is_exit": true,
+          "validator_index": 1,
+          "withdrawal_address": "0x2B888954421b424C5D3D9Ce9bB67c9bD47537d12"
       }
 
+   :resjson string entry_type: The type of the event that will be processed. Different validation is used based on the value for this field. Possible values are: ``"history event"``, ``"evm event"``, ``"eth withdrawal event"``, ``"eth block event"``, ``"eth deposit event"``.
    :reqjson string tx_hash: This is the transaction hash of the evm event
    :reqjson int sequence_index: This is an index that tries to provide the order of history entries for a single event_identifier.
    :reqjson int timestamp: The timestamp of the entry
```

Comment:
Don't forget to also split the arguments depending on the type. As it is now is unredeable